backend/src/models/user.model.ts

```diff
@@ -21,4 +22,9 @@ const UserSchema = new mongoose.Schema({
     }
 })
 
+UserSchema.pre('save', async function(){
+    const salt = await bcrypt.genSalt(10)
+    this.password = await bcrypt.hash(this.password, salt)
```

Comment:
If possible move this to util file

util.ts contains all reusable functions
ex: decryption, hashing, salting etc.